main.cpp

```diff
@@ -121,385 +163,563 @@ struct CarWash
  4) After you finish defining each type/function, click the [run] button.  Clear up any errors or warnings as best you can.
  */
 
+//UDT 1 // complete
 
-/*
-Thing 1) Laptop
-5 properties:
-    1) number of letter keys
-    2) number of numeric keys
-    3) amount of binary information in hard disk
-    4) number of usb ports
-    5) amount of electricity needed for portable use
-3 things it can do:
-    1) transfer information from usb device to hard disk
-    2) automatically adjust display brightness
-    3) provide update information to user
- */
 struct Laptop
 {
-	//number of letter keys
 	int numLetterKeys = 26;
-	//number of numeric keys
 	int numNumericKeys = 10;
-	//amount of binary information in hard disk
 	float binaryInfoInHardDisk = 499.96f;
-	//number of usb ports
 	int numUSBPorts = 2;
-	//amount of electricity needed for portable use
 	float electricityForPortableUse = 58.2f;
-	
-	struct User //first nested UDT
+
+	struct User
 	{
 		bool isUserMale = true;
 		int userAge = 37;
-		bool isUserAdmin = true;
+		bool isUserAdmin;
 		std::string userName = "Billy Crystal";
 		std::string password = "123456789";
+        int personalEmailTotal; // added member variable for nested 2nd implementation
+        int programmesOpen; // added member variable for nested 3rd implementation
 		
-		bool turnOnLaptop(bool laptopOff = true, bool userPressedPowerButton = true);
-		int sendEmails(int emailsPerDay = 10, bool personalEmails = true);
-		void openAProgram(int currentProgramAmount, bool isUserAdmin);
+		void turnOnLaptop(bool laptopOff, bool userPressedPowerButton);
+		auto sendEmails(bool personalEmails = true, int emailsPerDay = 10);
+		auto openAProgram(int currentProgramAmount, bool newProgrammeOpen);
 	};
```

Comment:
You've got some weird indentation here.

Reply:
weird indeed. Doesn't appear like this in replit. It seems to be common to any variables with comments beside them. I will delete and see